modules/codegen/src/smithy4s/codegen/internals/IR.scala

```diff
@@ -280,6 +280,8 @@ private[internals] object Hint {
     case object IndexedSeq extends SpecializedList
   }
   case object UniqueItems extends Hint
+  case class Typeclass(id: ShapeId, targetType: String, interpreter: String)
```

Comment:
A nice-to-have would be a type alias wherever a String refers to a Class (for doc purposes)